sys/dev/usb/net/if_urndis.c

```diff
@@ -641,7 +642,7 @@ urndis_ctrl_handle_reset(struct urndis_softc *sc,
 		msg_filter.filter = htole32(sc->sc_filter);
 
 		rval = urndis_ctrl_set(sc, OID_GEN_CURRENT_PACKET_FILTER,
-		    &msg_filter.hdr, sizeof(msg_filter));
```

Comment:
The commit log is no longer correct as it is now not just in attach.

On a larger note, I really don't like the bizarre nested union thing, and in particular it assumes the two different headers are the same size.  I think it would be cleaner to make it just be:

```
	static union {
		struct {
			struct rndis_query_req query;
			uint8_t eaddr[ETHER_ADDR_LEN];
		} eaddr;
		struct {
			struct rndis_set_req set;
			uint32_t filter;
		} filter;
	} msg;
```

Then you could just use `(struct <mumble> *)&msg.foo` and `sizeof(msg.foo)`.  I'm also not sure why this structure needs to be a global instead of on the stack.  Maybe silly fallout from swappable kernel stacks?

Reply:
Not obviously justified in https://github.com/CTSRD-CHERI/cheribsd/commit/fecf9642baeec1ab78378075ed2f492281efcd83

Reply:
Ewww, yeah.  I would say to cleanup the union upstream and drop static while at it.

Reply:
Ok. I'll make this change upstream first.

Reply:
This commit is now just a cherry-pick from upstream.